src/evaluate/evaluation_suite/__init__.py

```diff
@@ -0,0 +1,98 @@
+import importlib
+import inspect
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Callable, Optional, Union
+
+from datasets import Dataset, DownloadMode, load_dataset
+from datasets.utils.version import Version
+
+from evaluate.evaluator import evaluator
+from evaluate.loading import evaluation_module_factory
+from evaluate.utils.file_utils import DownloadConfig
+from evaluate.utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class SubTask:
```

Comment:
Did you end up with a better name for this class ?

Reply:
Thanks for reminding me! From the other PR thread, `EvaluationJob`, `EvaluationData` or `DataSource` were suggested.

I assume that you're referring to the tasks defined at https://huggingface.co/tasks — in which case, maybe SubTask is actually quite fitting? For example, the tasks listed under the [Text Classification task umbrella](https://huggingface.co/tasks/text-classification) are NLI, QNLI, MNLI, QQP — which right now correspond 1:1 to the SubTask object in EvaluationSuite (see an example of GLUE as an EvaluationSuite defined [here](https://huggingface.co/spaces/mathemakitten/glue-suite-v2/blob/main/glue-suite-v2.py)). 

A flip through NLP + CV lit suggests that "Task" is really the canonical name for this type of thing — see reference to NLP tasks in [Dynabench](https://ai.facebook.com/blog/dynabench-rethinking-ai-benchmarking/), [SuperGLUE](https://super.gluebenchmark.com/), the [Eleuther LM Harness](https://github.com/EleutherAI/lm-evaluation-harness), and computer vision as well like in [GRIT](https://allenai.org/project/grit/home). I'd love to keep convention with the field as much as possible so it's obvious to newcomers what the atomic unit of an EvaluationSuite should be. WDYT?